openmicroscopy/openmicroscopy
# LuceneQuerybuilder Mapannotations

This PR adds the colon character to the preserved characters when building the lucene query. This is needed for the specific MapAnnotation searches to work. On this occassion I also removed the LuceneQueryBuilder class and test from Insight, because the class has already been moved to the ome commons package.

To Test:
- Make sure that the search still works like before.
- Test the MapAnnotation specific searches: Add a Key-Value pair to a MapAnnotation, e. g. "foo"-"bar". Now add the term "foo" also to other MapAnnotations either as key or value and also to other kinds of Annotations (e. g. comment). Wait a bit for the search indexer been run. Now search for "foo:bar" and make sure you **only** see results which have the Key-Value pair "foo"-"bar" in the MapAnnotation. Then search for "has_key:foo" and make sure you **only** see results with a key "foo" in the MapAnnotation.
- Additional test: Because there was a merge conflict with #3506 , it would be good to check again, that connecting via Session ID still works.

--no-rebase
